cirq/circuits/qasm_output.py

```diff
@@ -311,7 +311,7 @@ def on_stuck(bad_op):
             should_annotate = decomposed != [main_op]
             if should_annotate:
                 output_line_gap(1)
-                if isinstance(main_op, ops.GateOperation):
+                if main_op.gate is not None:
```

Comment:
Might this be a problem if a new Operation is added that happens to have a gate subfield but is not a subtype of GateOperation? I would expect the default handling to pick up in this case.